### DISCUSSION TITLE
Extra instance

### DISCUSSION THREAD

**Discussion by lisadiez** [talmolab/sleap#1114] [Help!] [answered]
Hello!

We trained a top-down model to track wooden blocks in a home-cage paradigm. 

It's working very well but we noticed some frames where an extra instance is predicted:
<img width="1265" alt="blocks" src="https://user-images.githubusercontent.com/116014547/212136652-85405a70-bc48-4eca-b280-dc795d895cf7.png">

There are only two blocks in the cage, and the training set contained no more than two instances per frame - so I'm wondering why this is happening?

The incorrect (aka extra) prediction is usually the one with the lowest score, and I know there is an option to delete instances below a set threshold. But I'm hoping you could offer a better solution?

Thank you so much, as always!
Lisa

**Comment by roomrys** [comment, marked as answer]
Hi @lisadiez,

Culling instances during the prediction process (before predictions are even output as `Instance`s) is something that is on our radar. To do it correctly however, requires restructuring deep within our current inference pipeline. For the next release, we will have PR #1011 merged which allows setting a max number of instances for Top-Down models.

For now:

### CLI Option
You can run [`sleap-track`](https://sleap.ai/guides/cli.html?highlight=cli#sleap-track) with the `--peak_threshold` option set a bit higher.
#### Caveat
As @sean-afshar mentions in #1113, this option is only available through the CLI at the moment.

### GUI Option
The other option is to specify the `Target Instance Count` and `Cull to Target Instance Count` in the Inference Pipeline.
#### Caveat
This will add tracks to your project.

Thanks,
Liezl

**Reply by lisadiez** [reply]
This is very helpful, thank you Liezl :)
